File: examples-sdk/python/011-basic-graphql-api.py
#!/usr/bin/env python3
"""011-basic-graphql-api — Python SDK equivalent
Compile: vil compile --from python --input 011-basic-graphql-api.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("graphql-api", port=8080)
graphql = server.service_process("graphql")
graphql.endpoint("GET", "/", "index")
graphql.endpoint("GET", "/schema", "schema_info")
graphql.endpoint("GET", "/entities", "list_entities")
graphql.endpoint("POST", "/query", "graphql_query")
server.compile()

File: examples-sdk/typescript/011-basic-graphql-api.ts
#!/usr/bin/env tsx
// 011-basic-graphql-api — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 011-basic-graphql-api.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("graphql-api", 8080);
const graphql = new ServiceProcess("graphql");
graphql.endpoint("GET", "/", "index");
graphql.endpoint("GET", "/schema", "schema_info");
graphql.endpoint("GET", "/entities", "list_entities");
graphql.endpoint("POST", "/query", "graphql_query");
server.service(graphql);
server.compile();
